22.RaytracedAO/closestHit.comp

```diff
@@ -74,30 +73,33 @@ void main()
 			vec3 geomNormal;
 			const vec3 lastVxPos = load_positions(geomNormal,batchInstanceData,indices);
 
+			
 			const bool frontfacing = bool((batchInstanceData.determinantSignBit^floatBitsToUint(dot(normalizedV,geomNormal)))&0x80000000u);
-
+							
 			// get material
 			const nbl_glsl_MC_oriented_material_t material = nbl_glsl_MC_material_data_t_getOriented(batchInstanceData.material,frontfacing);
-			contrib.color = contrib.albedo = nbl_glsl_MC_oriented_material_t_getEmissive(material);
+
+			// if we ever support spatially varying emissive, we'll need to hoist barycentric computation and UV fetching to the position fetching
+			const vec2 compactBary = vec2(1.f-intersection.uv.x-intersection.uv.y,intersection.uv.x); // radeon rays is a special boy and does its barycentrics weird
+			
+			//
+			const nbl_glsl_xoroshiro64star_state_t scramble_start_state = nbl_glsl_xoroshiro64star_state_t(ray.mask,ray._active);
+			
+			//
+			normalizedN = load_normal_and_prefetch_textures(
+				batchInstanceData,indices,compactBary,geomNormal,material
+#ifdef TEX_PREFETCH_STREAM
+				,mat2(0.0) // TODO: Covariance Rendering
+#endif
+			);
```

Comment:
you don't need to hoist any of this, you already have normalizedV, that comment (you can clear it up) was about taking emissive from a texture via a UV mapping